### PR TITLE
Feat/mymagicai

### PR DESCRIPTION
**Pull request by vitali87**
# Description

MyMagic AI provides batch inference at scale. This PR integrates (wraps) MyMagic AI's API into existing LlamaIndex integrations. 

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Called the integrated MyMagic API with test user
- [ ] Added new notebook (that tests end-to-end)

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                )
+            time.sleep(poll_interval)
+
+    def stream_complete(self, question: str) -> CompletionResponseGen:
+        raise NotImplementedError(
+            "MyMagicAI does not currently support streaming completion."
+        )
+
+    async def achat(self, question: str) -> ChatResponse:
+        raise NotImplementedError("MyMagicAI does not currently support chat.")
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py:146]: Should we consider implementing chat and achat by converting the list of messages to a string? Other LLM classes support this (I.e. Llama.cpp)

File: llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    LLMMetadata,
+)
+from llama_index.core.llms.llm import LLM
+from llama_index.core.bridge.pydantic import Field
+
+
+class MyMagicAI(LLM):
+    base_url_template: str = "https://{model}.mymagic.ai"
+    api_key: str = None
+    model: str = Field(default="mistral7b", description="The MyMagicAI model to use.")
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py:22]: Do users need to worry about prompt formatting, or is mymgaic handling this in their API?

File: docs/examples/llm/mymagic.ipynb
Comment (logan-markewich) [inline docs/examples/llm/mymagic.ipynb:1]: The notebook has a few error outputs 

You can install rhe package with `pip install -e <path to root dir of mymgaic>` and then ensure your notebook venv us using the same env where you installed it.

If you can't get that to work, let's remove the notebook outputs altogether

Comment (vitali87) [reply docs/examples/llm/mymagic.ipynb:1]: Ok, I will try to make it work with venv and if it fails, I will remove outputs

File: llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py (hunk 1 above)
Comment (vitali87) [reply llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py:146]: I was thinking about it, actually.  Maybe we should, although we do not focus on chat. Do we have to do it now, or we can add it during the next update in a separate PR?

**Comment by logan-markewich** [comment]
I'm not able to contribute to this PR, but are you able to [install pants](https://www.pantsbuild.org/2.18/docs/getting-started/installing-pants) and run `pants tailor ::` ? This will insert the two missing BUILD files causing CI to fail

(We need to update the CLI package init tool to insert these 😅)

File: llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/llms/llama-index-llms-mymagic/llama_index/llms/mymagic/base.py:146]: definitely later, was just a thought :)

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**